runtime/backtrace_byt.c

```diff
@@ -296,22 +302,33 @@ void caml_stash_backtrace(value exn, value * sp, int reraise)
    updates *sp to point to the following one, and *trsp to the next
    trap frame, which we will skip when we reach it  */
 
-code_t caml_next_frame_pointer(value ** sp, value ** trsp)
+code_t caml_next_frame_pointer(value ** sp_inout, value ** trsp_inout)
 {
-  while (*sp < Caml_state->stack_high) {
-    value *spv = (*sp)++;
-    code_t *p;
-    if (Is_long(*spv)) continue;
-    p = (code_t*) spv;
-    if(&Trap_pc(*trsp) == p) {
-      *trsp = *trsp + Long_val(Trap_link_offset(*trsp));
+  value * sp = *sp_inout;
+  value * tr = *trsp_inout;
+  value v;
+  code_t p;
+
+  while (sp < Caml_state->stack_high) {
+    if (sp == tr) {
+      tr = tr + Long_val(Trap_link_offset(tr));
+      sp += 4;
```

Comment:
I got lost among the `*` so I tried to write equivalent code that would be clearer to me.   (Basically I prefer in-out parameters to by-reference parameters.) Perhaps I failed.  What did I change?

Reply:
I have to say that I don't find any of the two versions very clear. This is probably because the problem to solve itself is not particularly easy.

Anyway, in the original version, we did not increment any pointer by 4, so I am surprised to see this in this version.

Reply:
I should have commented.  The idea was to skip the whole trap frame when we hit it.  A trap frame is 4 words on the stack, and no code pointer can appear there.